### PR TITLE
factors: Add basic CI

### PR DESCRIPTION
**Pull request by lann**

### CODE REVIEW

File: crates/factor-variables/build.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 fn main() {
-    println!("cargo::rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-changed=build.rs");
```
Comment (rylev) [inline crates/factor-variables/build.rs:2]: If this line necessary? I believe I commented before that build.rs files automatically build if they're modified, but I'm not sure what the response to that was.

File: crates/key-value/src/host_component.rs [hunk 2]
```diff
@@ -56,7 +56,8 @@ impl HostComponent for KeyValueComponent {
     }
 }
 
-impl DynamicHostComponent for KeyValueComponent {
+// impl DynamicHostComponent for KeyValueComponent {
```
Comment (rylev) [inline crates/key-value/src/host_component.rs:59]: Any reason to keep this line around?

**Review by rylev — APPROVED** [review]

File: crates/factor-variables/build.rs (hunk 1 above)
Comment (rylev) [reply crates/factor-variables/build.rs:2]: Ah - I had somehow completely forgotten that that's the case...

**Comment by lann** [comment]
I gave up on getting spin-core tests passing in CI. YOU WIN THIS TIME GITHUB ACTIONS

**Pull request merged**